projects/core/src/cms/page/base-page-meta.resolver.ts

```diff
@@ -65,4 +69,62 @@ export class BasePageMetaResolver
   resolveRobots(): Observable<PageRobotsMeta[]> {
     return this.robots$;
   }
+
+  /**
+   * Resolves the canonical for the page.
+   *
+   * The canonical url is created by the help of the default `CanonicalUrlOptions` from
+   * the pageMeta options. The options can be further adjusted by the options argument.
+   */
+  resolveCanonicalUrl(
+    options?: CanonicalUrlOptions,
+    url?: string
+  ): Observable<string> {
+    return of(this.buildCanonicalUrl(options, url));
```

Comment:
Observable but not changing?
Does it even need to be a part of page meta model at all?

Reply:
Good point! I've operated on the `page$` observable by now.